### PR TITLE
Api 12293 remove extra validations

### PR DESCRIPTION
**Pull request by stiehlrod**
## Description of change
Look into the date validations across all three appeal types

## Original issue(s)
[API-12293](https://vajira.max.gov/browse/API-12293)

## Things to Know About This PR
Question posed: can we validate that a date is in the past with the schemas?

### Research Findings
It seems that many people have opened [issues](https://github.com/json-schema-org/json-schema-spec/issues/116) hoping for this [feature](https://github.com/dcs4cop/xcube/pull/336), and as far as I can tell, they have all been closed for various reasons. A somewhat recent [documentation](https://json-schema.org/draft/2019-09/json-schema-validation.html#rfc.section.7.3.1) states that json_)schema allows for time duration as part of the schema, but not a max value for a date.

### CODE REVIEW

File: modules/appeals_api/spec/config/schemas/v1/200996_headers_json_spec.rb [hunk 1]
```diff
@@ -12,10 +12,16 @@
   let(:errors) { validator.validate(json).to_a }
   let(:json) { fixture_as_json 'valid_200996_headers.json', version: 'v1' }
   let(:validator) { JSONSchemer.schema(json_schema) }
+  let(:header_errors) { validator.validate(invalid_headers).to_a }
```
Comment (ElleeB) [inline modules/appeals_api/spec/config/schemas/v1/200996_headers_json_spec.rb:15]: Were these changes left in by mistake?

File: modules/appeals_api/spec/fixtures/v2/invalid_200996_headers_characters.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
   "X-VA-First-Name": "J∑˙∆",
   "X-VA-Middle-Initial": "∂",
   "X-VA-Last-Name": "Smith",
-  "X-VA-Birth-Date": "1969-12-31",
+  "X-VA-Birth-Date": "not a date",
```
Comment (ElleeB) [inline modules/appeals_api/spec/fixtures/v2/invalid_200996_headers_characters.json:6]: Same as above, right?

File: modules/appeals_api/spec/models/higher_level_review_spec.rb [hunk 3]
```diff
@@ -183,15 +183,15 @@
...
-    context 'birth date isn\'t a date' do
-      let(:auth_headers) { default_auth_headers.merge 'X-VA-Birth-Date' => 'apricot' }
-
-      it 'using a birth date that isn\'t a date creates an invalid record' do
-        expect(higher_level_review.valid?).to be false
-        expect(higher_level_review.errors.to_a.length).to eq 1
-        expect(higher_level_review.errors.to_a.first.downcase).to include 'isn\'t a date'
-      end
-    end
+    # context 'birth date isn\'t a date' do
```
Comment (ElleeB) [inline modules/appeals_api/spec/models/higher_level_review_spec.rb:186]: Go ahead and permanently remove all tests related to the removed model validations.

File: modules/appeals_api/spec/config/schemas/v1/200996_headers_json_spec.rb (hunk 1 above)
Comment (stiehlrod) [reply modules/appeals_api/spec/config/schemas/v1/200996_headers_json_spec.rb:15]: yes, i forgot. Thank you

File: modules/appeals_api/spec/requests/v1/notice_of_disagreements_controller_spec.rb [hunk 4]
```diff
@@ -92,6 +92,35 @@ def base_path(path)
...
+      end
+    end
+
+    context 'returns 422 when decison date is not a date' do
+      let(:error_content) do
+        { 'status' => 422, 'detail' => "'banana' did not fit within the defined length limits" }
+      end
+
+      it 'when given a string for the contestable issues decision date ' do
+        data = JSON.parse(@data)
```
Comment (ElleeB) [inline modules/appeals_api/spec/requests/v1/notice_of_disagreements_controller_spec.rb:114]: I think should be able to use the already defined `parse` here

File: modules/appeals_api/spec/requests/v2/higher_level_reviews_controller_spec.rb [hunk 5]
```diff
@@ -40,7 +40,7 @@ def base_path(path)
       it 'creates an HLR and persists the data' do
         post(path, params: @data, headers: @minimum_required_headers)
         expect(parsed['data']['type']).to eq('higherLevelReview')
-        expect(parsed['data']['attributes']['status']).to eq('pending')
```
Comment (ElleeB) [inline modules/appeals_api/spec/requests/v2/higher_level_reviews_controller_spec.rb:43]: Definitely want to revert this

File: modules/appeals_api/spec/requests/v2/notice_of_disagreements_controller_spec.rb [hunk 6]
```diff
@@ -49,6 +49,36 @@ def base_path(path)
...
+        @headers.merge!('X-VA-Birth-Date' => 'apricot')
+        post(path, params: @data.to_json, headers: @headers)
+        expect(response.status).to eq(422)
+        expect(parsed['errors']).to be_an Array
+      end
+    end
+
+    context 'returns 422 when decison date is not a date' do
+      let(:error_content) do
+        { 'status' => 422, 'detail' => "'banana' did not fit within the defined length limits" }
```
Comment (ElleeB) [inline modules/appeals_api/spec/requests/v2/notice_of_disagreements_controller_spec.rb:67]: Hm. Seems like the detail should be something like, "'banana' is not a date", right?

File: modules/appeals_api/spec/requests/v2/notice_of_disagreements_controller_spec.rb [hunk 7]
```diff
@@ -49,6 +49,36 @@ def base_path(path)
...
+
+      it 'when given a string for the contestable issues decision date ' do
+        data = JSON.parse(@max_data)
+        data['included'][0]['attributes'].merge!('decisionDate' => 'banana')
+
+        post(path, params: data.to_json, headers: @headers)
+        expect(response.status).to eq(422)
+        expect(parsed['errors']).to be_an Array
+        expect(parsed['errors'][0]['title']).to include('Invalid')
+        expect(parsed['errors'][0]['detail']).to include("'banana' did not fit")
```
Comment (ElleeB) [inline modules/appeals_api/spec/requests/v2/notice_of_disagreements_controller_spec.rb:78]: Same here...the detail should be something related to 'banana' not being a date

File: modules/appeals_api/spec/requests/v2/supplemental_claims_controller_spec.rb [hunk 8]
```diff
@@ -85,6 +85,37 @@ def base_path(path)
...
+
+      it 'when given a string for the contestable issues decision date ' do
+        sc_data = JSON.parse(data)
+        sc_data['included'][0]['attributes'].merge!('decisionDate' => 'banana')
+
+        post(path, params: sc_data.to_json, headers: headers)
+        expect(response.status).to eq(422)
+        expect(parsed['errors']).to be_an Array
+        expect(parsed['errors'][0]['title']).to include('Invalid')
+        expect(parsed['errors'][0]['detail']).to include("'banana' did not fit within the defined length limits")
```
Comment (ElleeB) [inline modules/appeals_api/spec/requests/v2/supplemental_claims_controller_spec.rb:115]: Same here...is this really what the error returns when submitting something other than a proper date? We'll need to fix this if that's the case!

File: modules/appeals_api/spec/requests/v2/notice_of_disagreements_controller_spec.rb (hunk 6 above)
Comment (ElleeB) [reply modules/appeals_api/spec/requests/v2/notice_of_disagreements_controller_spec.rb:67]: Note: We're going to create a ticket to look into whether this is the expected detail.

**Pull request merged**